--- a/Code/gml_GlobalScript_scr_moveheart.gml
+++ b/Code/gml_GlobalScript_scr_moveheart.gml
@@ -1,5 +1,12 @@
 function scr_moveheart()
 {
+    if (instance_exists(obj_herokris))
+    {
-    return instance_create(obj_herokris.x + 10, obj_herokris.y + 40, obj_moveheart);
+        return instance_create(obj_herokris.x + 10, obj_herokris.y + 40, obj_moveheart);
+    }
+    else
+    {
+        return instance_create(0, 0, obj_moveheart);
+    }
 }
 
